File: src/Notify.DeviceApi/Devices/DeviceRegistration.cs
using System.Text.Json.Serialization;

namespace Notify.DeviceApi.Devices;

// Registration request from the iOS app. The platform field is fixed to
// "apns" for v1 — Android can land later as a sibling enum value.
public sealed record DeviceRegistration
{
    [JsonPropertyName("deviceToken")]
    public required string DeviceToken { get; init; }

    [JsonPropertyName("platform")]
    public required string Platform { get; init; }

    [JsonPropertyName("tags")]
    public IReadOnlyList<string>? Tags { get; init; }
}

File: app/Notify/Models/DeviceRegistration.swift
import Foundation

// Request body for `POST /v1/devices`. Mirrors src/Notify.Functions/Devices/DeviceRegistration.cs.
// Platform is fixed to "apns" for v1; Android can be a sibling case later.
struct DeviceRegistration: Codable, Equatable {
    let deviceToken: String
    let platform: String
    let tags: [String]?

    init(apnsToken: String, tags: [String]? = nil) {
        self.deviceToken = apnsToken
        self.platform = "apns"
        self.tags = tags
    }
}

// Response from `POST /v1/devices` on the 202 path.
struct DeviceRegistrationResponse: Decodable, Equatable {
    let installationId: String
}
